### ISSUE TITLE
Link to correct line when editing a translation

### ISSUE THREAD

**Issue by mgeisler** [google/comprehensive-rust#1075]
On a whim, I tried adding a `#Lnnn` fragment to a URL which opens the GitHub code editor... and it works! Check out [line 3456 of the Korean translation](https://github.com/google/comprehensive-rust/edit/main/po/ko.po#L3456).

With this, we are able to create useful links on the published pages for the translations. Currently, we simply link to the `xx.po` file, but we should instead link to the first message of each page. We could do so in a few different ways:

- We could post-process the HTML. Something like loading each page, find the first paragraph, look it up in the `xx.po` file, get the line number, inject the line number into the link on the page.
- We could use the GitHub API to download the file on demand when clicking the edit link. We would then search like above and send people to the right spot in the editor. The PO files are 500-800 KB in size, so that might be a little heavy-handed.

We could go with a hybrid version: generate and publish a small `xx.json` file with the correct line number for every path. Then download this file when clicking the link. That ought to be fairly simple since it's easy to load JSON files from JavaScript, and it will be pretty simple to produce this file when we publish the course.

Implementing this would drastically improve the editing experience for our translators.

**Comment by ishqDehlvi** [comment]
How can I reproduce this issue??

**Comment by mgeisler** [comment]
> How can I reproduce this issue??

Try clicking the little pencil in the top-right corner of a translation. It will take you to the `xx.po` file, but it won't jump to the relevant line in the file.

**Comment by mgeisler** [comment]
Try with this page: https://google.github.io/comprehensive-rust/pt-BR/running-the-course.html

**Comment by ishqDehlvi** [comment]
Is this that thing?? 

Uploading Screen Recording 2023-11-02 at 8.05.00 AM.mov…

**Comment by mgeisler** [comment]
Yes, precisely! If you use the pencil on one of the English pages, then you're taking to the correct Markdown file in GitHub. However, if you do it on a translated page (like in your video), then you're taking to the huge PO file instead. That's not very useful for people who want to submit fixes to the translations.

**Comment by ishqDehlvi** [comment]
Got it lemme figure out something and raise a pr for this issue... Thanks for clarification
